### PR TITLE
feat: Handle Snaps markdown in the notification ui

### PR DESCRIPTION
**Pull request by matteoscurati**
## **Description**

This PR introduces the component for Snap notifications integrated into the current UI. This PR allows for the use of markdown as specified by the current Snap notifications.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/24738?quickstart=1)

## **Screenshots/Recordings**

https://www.loom.com/share/adc72211336745869263e59904700ebd

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ui/components/app/snaps/snap-ui-markdown/snap-ui-markdown.js [hunk 1]
```diff
@@ -23,7 +23,8 @@ const Paragraph = (props) => (
     variant={TextVariant.bodyMd}
     className="snap-ui-markdown__text"
     overflowWrap={OverflowWrap.Anywhere}
-    color={TextColor.inherit}
+    color={TextColor.textDefault}
+    textVariant={TextVariant.bodyMd}
```
Comment (GuillaumeRx) [inline ui/components/app/snaps/snap-ui-markdown/snap-ui-markdown.js:27]: Not sure what's the utility of `textVariant` on the `Text` element

File: ui/components/app/snaps/snap-ui-markdown/snap-ui-markdown.js [hunk 2]
```diff
@@ -23,7 +23,8 @@ const Paragraph = (props) => (
     variant={TextVariant.bodyMd}
     className="snap-ui-markdown__text"
     overflowWrap={OverflowWrap.Anywhere}
-    color={TextColor.inherit}
+    color={TextColor.textDefault}
```
Comment (GuillaumeRx) [inline ui/components/app/snaps/snap-ui-markdown/snap-ui-markdown.js:26]: Why can't we inherit anymore ?

Comment (FrederikBolding) [reply ui/components/app/snaps/snap-ui-markdown/snap-ui-markdown.js:26]: We need to inherit to support certain cases where the color is overridden by the parent

File: app/images/snaps-icon.svg
Comment (FrederikBolding) [inline app/images/snaps-icon.svg:1]: This is not in use

File: ui/components/app/snaps/snap-ui-markdown/snap-ui-markdown.js (hunk 1 above)
Comment (FrederikBolding) [inline ui/components/app/snaps/snap-ui-markdown/snap-ui-markdown.js:27]:
```suggestion
```

This is already applied via `variant` AFAIK

File: ui/components/multichain/notification-list-snap-button/notification-list-snap-button.tsx
Comment (FrederikBolding) [inline ui/components/multichain/notification-list-snap-button/notification-list-snap-button.tsx:1]: Isn't this unused now?

File: app/images/snaps-icon.svg
Comment (matteoscurati) [reply app/images/snaps-icon.svg:1]: @FrederikBolding you are right. Reverted!

File: ui/components/app/snaps/snap-ui-markdown/snap-ui-markdown.js (hunk 2 above)
Comment (matteoscurati) [reply ui/components/app/snaps/snap-ui-markdown/snap-ui-markdown.js:26]: Reverted!

File: ui/components/app/snaps/snap-ui-markdown/snap-ui-markdown.js (hunk 1 above)
Comment (matteoscurati) [reply ui/components/app/snaps/snap-ui-markdown/snap-ui-markdown.js:27]: Reverted!

File: ui/components/multichain/notification-list-item-snap/notification-list-item-snap.tsx [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+import { SnapUIMarkdown } from '../../app/snaps/snap-ui-markdown';
+import SnapAvatar from '../../app/snaps/snap-avatar';
+
+export type NotificationListItemSnapProps = {
+  id: string;
+  isRead: boolean;
+  title: NotificationListItemTextProps;
+  snapMessage: string;
+  createdAt: Date;
+  amount?: string;
```
Comment (FrederikBolding) [inline ui/components/multichain/notification-list-item-snap/notification-list-item-snap.tsx:29]:
```suggestion
```

File: ui/components/multichain/notification-list-item-snap/notification-list-item-snap.tsx
Comment (FrederikBolding) [inline ui/components/multichain/notification-list-item-snap/notification-list-item-snap.tsx:1]: Screenshot 2024-05-23 at 17 34 02
Alignment is off on this component compared to Figma. Can we align the icon with the text?

File: ui/components/app/snaps/snap-ui-markdown/snap-ui-markdown.js (hunk 1 above)
Comment (matteoscurati) [reply ui/components/app/snaps/snap-ui-markdown/snap-ui-markdown.js]: You are right. I removed this one

File: ui/components/multichain/notification-list-item-snap/notification-list-item-snap.tsx [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+        />
+
+        <Box
+          display={Display.Flex}
+          gap={4}
+          height={BlockSize.Full}
+          alignItems={AlignItems.flexStart}
+          width={BlockSize.Full}
+        >
+          {/* <Box padding={4} /> */}
```
[the remark this answers is not in the document]
Comment (matteoscurati) [reply ui/components/multichain/notification-list-item-snap/notification-list-item-snap.tsx:119]: Ups :)

File: ui/components/multichain/notification-list-item-snap/notification-list-item-snap.tsx
Comment (matteoscurati) [reply ui/components/multichain/notification-list-item-snap/notification-list-item-snap.tsx:1]: Done! Good catch!

**Review by FrederikBolding — DISMISSED** [review]

**Review by FrederikBolding — DISMISSED** [review]

**Review by Prithpal-Sooriya — DISMISSED** [review]

**Review by FrederikBolding — APPROVED** [review]

**Review by Prithpal-Sooriya — APPROVED** [review]

**Pull request merged**